src/input/element-input.js

```diff
@@ -844,9 +872,10 @@ class ElementInput {
                 ray = ray3d;
             }
 
-            if (ray && this._checkElement(ray, element, screen)) {
+            var d = this._checkElement(ray, element, screen);
+            if (ray && d > 0 && d < closestDistance) {
                 result = element;
-                break;
+                closestDistance = d;
```

Comment:
I have a feeling this code changes the 2d path and not only 3d path I thought should return the nearest one. The 2d path should return the first one that it hits?

Reply:
Fixed in next commit.